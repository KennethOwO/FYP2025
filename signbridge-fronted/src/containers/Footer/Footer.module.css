.footer_container {
    width: 100%;
    background-color: #1c2e4a;
    padding: 2.5rem 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin-top: auto;
}

.footer_links {
    width: 100%;
    max-width: 1200px;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
}

.footer_link_wrapper {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
}

.footer_link_items {
    margin: 10px 10px;
    padding-left: 20px;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    text-align: left;
    width: 180px;
    min-width: 200px;
    box-sizing: border-box;
}

.footer_link_items h2 {
    margin-bottom: 16px;
}

.footer_link_items > h2 {
    color: #fff;
}

.footer_link_items a {
    color: #fff;
    text-decoration: none;
    margin-bottom: 0.5rem;
}

.footer_link_items a:hover {
    color: #e9e9e9;
    transition: 0.3s ease-out;
}

/* Social Icons */
.social-icon-link {
    color: #fff;
    font-size: 24px;
}

.social_media {
    max-width: 1000px;
    width: 100%;
}

.social_media_wrap {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 90%;
    max-width: 1000px;
    margin: 40px auto 0 auto;
}

.social_icons {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 240px;
}

.website_rights {
    color: #fff;
    margin-bottom: 16px;
}

.footer_icon {
    color: white;
    font-size: 24px;
}

@media screen and (max-width: 820px) {
    .footer_link_wrapper {
        flex-direction: row;
    }

    .footer_link_items {
        margin: 10px 10px;
        width: 150px;
    }

    .social_media_wrap {
        flex-direction: column;
        gap: 20px;
    }

    .social_icons {
        width: 100%;
        justify-content: center;
        gap: 20px;
    }

    .website_rights {
        text-align: center;
        margin-bottom: 0;
    }
}

@media screen and (max-width: 480px) {
    .footer_link_items {
        min-width: 50px;
    }
}
