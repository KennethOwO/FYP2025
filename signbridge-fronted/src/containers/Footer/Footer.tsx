import { Link } from "react-router-dom";
import { useThemeStore } from "../../store/theme";
import styles from "./Footer.module.css";
import { useTranslation } from "react-i18next";

function Footer() {
    const { color } = useThemeStore();
    const { t } = useTranslation();

    return (
        <div className={styles.footer_container} style={{ background: color }}>
            <div className={styles.footer_links}>
                <div className={styles.footer_link_wrapper}>
                    <div className={styles.footer_link_items}>
                        <h2>Sign Bridge</h2>
                        <Link to="/under-construction">{t("how_it_works")}</Link>
                        <Link to="/under-construction">{t("testimonials")}</Link>
                        <Link to="/under-construction">{t("careers")}</Link>
                        <Link to="/under-construction">{t("investors")}</Link>
                        <Link to="/under-construction">{t("terms_of_service")}</Link>
                        <Link to="/references">{t("references")}</Link>
                    </div>
                    <div className={styles.footer_link_items}>
                        <h2>{t("contact_us")}</h2>
                        <Link to="/under-construction">{t("contact")}</Link>
                        <Link to="/under-construction">{t("support")}</Link>
                        <Link to="/under-construction">{t("destinations")}</Link>
                        <Link to="/under-construction">{t("sponsorship")}</Link>
                    </div>
                </div>
                <div className={styles.footer_link_wrapper}>
                    <div className={styles.footer_link_items}>
                        <h2>{t("modules")}</h2>
                        <Link to="/library">{t("library")}</Link>
                        <Link to="/communication">{t("communication")}</Link>
                        <Link to="/education">{t("education")}</Link>
                    </div>
                    <div className={styles.footer_link_items}>
                        <h2>{t("social_media")}</h2>
                        <Link to="https://www.facebook.com/neuon.ai.my" target="_blank" rel="noopener noreferrer">
                            Facebook
                        </Link>
                        <Link to="https://www.linkedin.com/company/neuon-ai/" target="_blank" rel="noopener noreferrer">
                            LinkedIn
                        </Link>
                        <Link to="https://www.youtube.com/@neuonai3999" target="_blank" rel="noopener noreferrer">
                            Youtube
                        </Link>
                        <Link to="https://github.com/neuonai" target="_blank" rel="noopener noreferrer">
                            GitHub
                        </Link>
                        <Link to="https://www.pinterest.com/neuon_ai/_created/" target="_blank" rel="noopener noreferrer">
                            Pinterest
                        </Link>
                    </div>
                </div>
            </div>
            <section className={styles.social_media}>
                <div className={styles.social_media_wrap}>
                    <small className={styles.website_rights}>{t("copyright")}</small>
                    <div className={styles.social_icons}>
                        <Link className={styles.social_icon_link} to="https://www.facebook.com/neuon.ai.my" target="_blank" aria-label="Facebook">
                            <i className={`fab fa-facebook-f ${styles.footer_icon}`} />
                        </Link>
                        <Link className={styles.social_icon_link} to="https://www.linkedin.com/company/neuon-ai/" target="_blank" aria-label="LinkedIn">
                            <i className={`fab fa-linkedin ${styles.footer_icon}`} />
                        </Link>
                        <Link className={styles.social_icon_link} to="https://www.youtube.com/@neuonai3999" target="_blank" aria-label="Youtube">
                            <i className={`fab fa-youtube ${styles.footer_icon}`} />
                        </Link>
                        <Link className={styles.social_icon_link} to="https://github.com/neuonai" target="_blank" aria-label="GitHub">
                            <i className={`fa-brands fa-github ${styles.footer_icon}`} />
                        </Link>
                        <Link className={styles.social_icon_link} to="https://www.pinterest.com/neuon_ai/_created/" target="_blank" aria-label="Twitter">
                            <i className={`fa-brands fa-pinterest ${styles.footer_icon}`} />
                        </Link>
                    </div>
                </div>
            </section>
        </div>
    );
}

export default Footer;
