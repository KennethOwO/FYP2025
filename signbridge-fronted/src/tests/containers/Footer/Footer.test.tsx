import React from "react";
import Footer from "@root/containers/Footer/Footer";
import { create, act } from "react-test-renderer";
import { cleanup } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key, // Mock the translation function to return the key
    i18n: {
      changeLanguage: jest.fn(), // Mock the changeLanguage function
    },
  }),
}));

describe("Test Footer", () => {
  afterEach(() => {
    // Clean up the DOM
    cleanup();
    // Reset mocks to their initial state
    jest.resetAllMocks();
  });

  it("should render correctly", () => {
    // Mock the document.cookie
    Object.defineProperty(document, "cookie", {
      writable: true,
      value: "token=abc123; otherCookie=someValue",
    });

    let tree;
    act(() => {
      tree = create(
        <MemoryRouter>
          <Footer />
        </MemoryRouter>
      );
    });

    expect(tree!.toJSON()).toMatchSnapshot();
  });
});
