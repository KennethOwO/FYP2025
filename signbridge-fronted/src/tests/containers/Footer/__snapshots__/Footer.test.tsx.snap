// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Footer should render correctly 1`] = `
<div
  className="footer-container"
  style={
    {
      "background": "#1C2E4A",
    }
  }
>
  <div
    className="footer-links"
  >
    <div
      className="footer-link-wrapper"
    >
      <div
        className="footer-link-items"
      >
        <h2>
          Sign Bridge
        </h2>
        <a
          href="/login"
          onClick={[Function]}
        >
          how_it_works
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          testimonials
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          careers
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          investors
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          terms_of_service
        </a>
        <a
          href="/references"
          onClick={[Function]}
        >
          references
        </a>
      </div>
      <div
        className="footer-link-items"
      >
        <h2>
          contact_us
        </h2>
        <a
          href="/"
          onClick={[Function]}
        >
          contact
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          support
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          destinations
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          sponsorship
        </a>
      </div>
    </div>
    <div
      className="footer-link-wrapper"
    >
      <div
        className="footer-link-items"
      >
        <h2>
          modules
        </h2>
        <a
          href="/library"
          onClick={[Function]}
        >
          library
        </a>
        <a
          href="/communication"
          onClick={[Function]}
        >
          communication
        </a>
        <a
          href="/education"
          onClick={[Function]}
        >
          education
        </a>
      </div>
      <div
        className="footer-link-items"
      >
        <h2>
          social_media
        </h2>
        <a
          href="/"
          onClick={[Function]}
        >
          Instagram
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          Facebook
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          Youtube
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          Twitter
        </a>
      </div>
    </div>
  </div>
  <section
    className="social-media"
  >
    <div
      className="social-media-wrap"
    >
      <small
        className="website-rights"
      >
        copyright
      </small>
      <div
        className="social-icons"
      >
        <a
          aria-label="Facebook"
          className="social-icon-link facebook"
          href="/"
          onClick={[Function]}
          target="_blank"
        >
          <i
            className="fab fa-facebook-f"
          />
        </a>
        <a
          aria-label="Instagram"
          className="social-icon-link instagram"
          href="/"
          onClick={[Function]}
          target="_blank"
        >
          <i
            className="fab fa-instagram"
          />
        </a>
        <a
          aria-label="Youtube"
          className="social-icon-link youtube"
          href="/"
          onClick={[Function]}
          target="_blank"
        >
          <i
            className="fab fa-youtube"
          />
        </a>
        <a
          aria-label="Twitter"
          className="social-icon-link twitter"
          href="/"
          onClick={[Function]}
          target="_blank"
        >
          <i
            className="fab fa-twitter"
          />
        </a>
        <a
          aria-label="LinkedIn"
          className="social-icon-link twitter"
          href="/"
          onClick={[Function]}
          target="_blank"
        >
          <i
            className="fab fa-linkedin"
          />
        </a>
      </div>
    </div>
  </section>
</div>
`;
